web/src/main/java/org/springframework/security/web/server/ServerHttpBasicAuthenticationConverter.java

```diff
@@ -51,9 +52,8 @@
 		if (!StringUtils.startsWithIgnoreCase(authorization, "basic ")) {
 			return Mono.empty();
 		}
-		String credentials = (authorization.length() <= BASIC.length()) ? ""
-				: authorization.substring(BASIC.length(), authorization.length());
-		String decoded = new String(base64Decode(credentials));
+		String credentials = (authorization.length() <= BASIC.length()) ? "" : authorization.substring(BASIC.length());
+		String decoded = new String(base64Decode(credentials), StandardCharsets.UTF_8);
```

Comment:
Based on discussion in the parent issue, I think I'm now leaning towards actually making this configurable with a default. This would have the benefit of matching up well with the servlet implementation. Would you mind adding that option and a few corresponding tests with for example `ISO_8859_1`?